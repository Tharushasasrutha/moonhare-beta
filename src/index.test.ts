import Processor from '.'
import CleanCSS from 'clean-css'

const shuffle = (array: string[]) => {
    for (let i = array.length - 1; i > 0; i--) {
        const j = Math.floor(Math.random() * (i + 1))
        ;[array[i], array[j]] = [array[j], array[i]]
    }
    return array
}

describe('Processor', () => {
    const processor = new Processor()

    test('sample', () => {
        const run = () => {
            const processor = new Processor()
            const {styleSheet} = processor.interpret(
                shuffle([
                    'container',
                    'decoration-slice',
                    'box-border',
                    'clear-left',
                    '@dark:float-right',
                    'isolate',
                    'object-cover',
                    'overflow-hidden',
                    'static',
                    '!-z-30',
                    '-!inset-4'
                ]).join('    ')
            )

            expect(
                new CleanCSS({format: 'beautify'}).minify(styleSheet.build())
                    .styles
            ).toMatchInlineSnapshot(`
                ".container {
                  width: 100%
                }
                @media (min-width:640px) {
                  .container {
                    width: 640px
                  }
                }
                @media (min-width:768px) {
                  .container {
                    width: 768px
                  }
                }
                @media (min-width:1024px) {
                  .container {
                    width: 1024px
                  }
                }
                @media (min-width:1280px) {
                  .container {
                    width: 1280px
                  }
                }
                @media (min-width:1536px) {
                  .container {
                    width: 1536px
                  }
                }
                .decoration-slice {
                  box-decoration-break: slice
                }
                .box-border {
                  box-sizing: border-box
                }
                .@dark:float-right {
                  float: right
                }
                .clear-left {
                  clear: left
                }
                .isolate {
                  isolation: isolate
                }"
            `)
        }
        run()
        run()
    })
})
