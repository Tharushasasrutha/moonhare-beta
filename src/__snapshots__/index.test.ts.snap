// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Processor sample 1`] = `
".container {
  width: 100%
}
@media (min-width:640px) {
  .container {
    width: 640px
  }
}
@media (min-width:768px) {
  .container {
    width: 768px
  }
}
@media (min-width:1024px) {
  .container {
    width: 1024px
  }
}
@media (min-width:1280px) {
  .container {
    width: 1280px
  }
}
@media (min-width:1536px) {
  .container {
    width: 1536px
  }
}
.decoration-slice {
  box-decoration-break: slice
}
.box-border {
  box-sizing: border-box
}
.clear-left {
  clear: left
}
.object-cover {
  object-position: cover
}
.overflow-hidden {
  overflow: hidden
}
.static {
  position: static
}
.-\\\\!inset-4 {
  top: -1rem!important;
  right: -1rem!important;
  bottom: -1rem!important;
  left: -1rem!important
}
.\\\\!-z-30 {
  z-index: -30!important
}
.isolate {
  isolation: isolate
}
@media (prefers-color-scheme:dark) {
  .\\\\@dark\\\\:float-right {
    float: right
  }
}"
`;
